src/script/descriptor.cpp

```diff
@@ -1515,6 +1515,9 @@ std::vector<std::unique_ptr<PubkeyProvider>> ParsePubkeyInner(uint32_t key_exp_i
     if (str.size() == 0) {
         error = "No key provided";
         return {};
+    } else if (IsSpace(str[0]) || IsSpace(str.back())) {
+        error = strprintf("Pubkey '%s' is invalid due to whitespace", str);
+        return {};
```

Comment:
nit: Because there is a `return` statement in the if block that is above, you could remove the `else` and use `str.front()` instead of `str[0]`.

Reply:
Done.